Networking/Networking/Model/MetaData.swift

```diff
@@ -39,18 +47,92 @@ private extension MetaData {
     }
 }
 
-private extension AnyCodable {
-    func getAsString() -> String {
-        switch value {
-        case let string as String:
-            return string
-        default:
-            if JSONSerialization.isValidJSONObject(value) {
-                let data = (try? JSONSerialization.data(withJSONObject: value)) ?? Data()
-                return String(data: data, encoding: .utf8) ?? ""
-            } else {
-                return "\(value)"
+public struct MetaDataValue: Codable, Equatable, Sendable {
+    private let valueHolder: ValueHolder
+
+    public var stringValue: String {
+        valueHolder.stringValue
+    }
+    public var rawValue: String {
+        valueHolder.rawValue
+    }
+
+    public var isJson: Bool {
+        switch valueHolder {
+        case .string:
+            return false
+        case .json:
+            return true
+        }
+    }
+
+    public init(rawValue: String) {
+        let data = Data(rawValue.utf8)
+        if let jsonObject = try? JSONDecoder().decode(AnyCodable.self, from: data) {
+            valueHolder = try! ValueHolder(from: jsonObject.value)
```

Comment:
We generally want to avoid force unwrapping: https://github.com/woocommerce/woocommerce-ios/blob/trunk/docs/coding-style-guide.md#forced-downcasts-and-unwrapping

Reply:
I may be missing something, so please bear with me here 😅.

I don't think I'm using any downcasting or unwrapping here, the `try!` is needed just to make the compiler happy as the `ValueHolder.init` is marked as `throws`: This should never crash, as the only case where `ValueHolder.init` could throw is it fails to encode the content as JSON, but since we confirm that the content has a valid JSON here, it should never throw.

Reply:
What @selanthiraiyan [said here](https://github.com/woocommerce/woocommerce-ios/pull/14292/files#r1837494558) was what I had in mind.

Reply:
I applied that change, thank you both.

But just for transparency, and as I shared with @selanthiraiyan, I personally prefer to use `try!` here, as having a crash (if for example the `ValueHoder.init` was modifier later) is better than using incorrect data.